tests/codex/testnode.nim

```diff
@@ -163,3 +195,112 @@ asyncchecksuite "Test Node":
+    discovery: DiscoveryEngine
+    manifest: Manifest
+    manifestCid: string
+
+  proc fetch(T: type Manifest, chunker: Chunker): Future[Manifest] {.async.} =
+    # Collect blocks from Chunker into Manifest
+    await storeDataGetManifest(localStore, chunker)
+
+  setup:
+    file = open(path.splitFile().dir /../ "fixtures" / "test.jpg")
+    chunker = FileChunker.new(file = file, chunkSize = DefaultBlockSize)
+    switch = newStandardSwitch()
+    wallet = WalletRef.new(EthPrivateKey.random())
+    network = BlockExcNetwork.new(switch)
+
+    clock = MockClock.new()
+    localStoreMetaDs = SQLiteDatastore.new(Memory).tryGet()
+    localStoreRepoDs = SQLiteDatastore.new(Memory).tryGet()
+    localStore = RepoStore.new(localStoreRepoDs, localStoreMetaDs, clock=clock)
+    await localStore.start()
+
+    blockDiscovery = Discovery.new(
+      switch.peerInfo.privateKey,
+      announceAddrs = @[MultiAddress.init("/ip4/127.0.0.1/tcp/0")
+        .expect("Should return multiaddress")])
+    peerStore = PeerCtxStore.new()
+    pendingBlocks = PendingBlocksManager.new()
+    discovery = DiscoveryEngine.new(localStore, peerStore, network, blockDiscovery, pendingBlocks)
+    engine = BlockExcEngine.new(localStore, wallet, network, discovery, peerStore, pendingBlocks)
+    store = NetworkStore.new(engine, localStore)
+    node = CodexNodeRef.new(switch, store, engine, nil, blockDiscovery) # TODO: pass `Erasure`
+
+    # Setup Host Contracts and dependencies
+    let market = MockMarket.new()
+    sales = Sales.new(market, clock, localStore, 0)
+    let hostContracts = some HostInteractions.new(clock, sales)
+    node.contracts = (ClientInteractions.none, hostContracts, ValidatorInteractions.none)
+
+    await node.start()
+
+    # Populate manifest in local store
+    manifest = await storeDataGetManifest(localStore, chunker)
+    let manifestBlock = bt.Block.new(
+        manifest.encode().tryGet(),
+        codec = DagPBCodec
+      ).tryGet()
+    manifestCid = $(manifestBlock.cid)
+    (await localStore.putBlock(manifestBlock)).tryGet()
+
+  teardown:
+    close(file)
+    await node.stop()
+
+  test "onExpiryUpdate callback":
+    let
+      # The blocks have set default TTL, so in order to update it we have to have larger TTL
+      expectedExpiry: SecondsSince1970 = clock.now + DefaultBlockTtl.seconds + 11123
+
+    without expiryUpdateCallback =? sales.onExpiryUpdate:
+      raiseAssert "onExpiryUpdate not set"
+
+    (await expiryUpdateCallback(manifestCid, expectedExpiry)).tryGet()
+
+    for index in 0..<manifest.blocksCount:
+      let blk = (await localStore.getBlock(manifest.treeCid, index)).tryGet
+      let expiryKey = (createBlockExpirationMetadataKey(blk.cid)).tryGet
+      let expiry = await localStoreMetaDs.get(expiryKey)
+
+      check (expiry.tryGet).toSecondsSince1970 == expectedExpiry
+
+  test "onStore callback":
+    without onStore =? sales.onStore:
+      raiseAssert "onStore not set"
```

Comment:
Same here, I'd make it a separate test to check that the onStore callback is set